Source/WebCore/loader/CookieJar.cpp

```diff
@@ -183,10 +183,10 @@ bool CookieJar::getRawCookies(const Document& document, const URL& url, Vector<C
 {
     auto pageID = document.pageID();
     std::optional<FrameIdentifier> frameID;
-    if (auto* frame = document.frame())
+    if (RefPtr frame = document.frame())
         frameID = frame->loader().frameID();
 
-    if (auto* session = m_storageSessionProvider->storageSession())
+    if (WeakPtr session = Ref { m_storageSessionProvider }->storageSession())
```

Comment:
We should probably make NetworkStorageSession inherit from CanMakeCheckedPtr and use a CheckedPtr here instead. `WeakPtr` doesn't prevent UAF here because even if WeakPtr was not null at the time of if condition evaluation, it can become null later underneath us. CheckedPtr will prevent UAF in that case.